staging/src/k8s.io/apiserver/pkg/server/filters/priority-and-fairness.go

```diff
@@ -356,3 +368,61 @@ func tooManyRequests(req *http.Request, w http.ResponseWriter, retryAfter string
 	w.Header().Set("Retry-After", retryAfter)
 	http.Error(w, "Too many requests, please try again later.", http.StatusTooManyRequests)
 }
+
+func newRequestWaitContextFactory(clock utilsclock.PassiveClock, defaultRequestWaitLimit time.Duration) *requestWaitContextFactory {
+	return &requestWaitContextFactory{
+		clock:                   clock,
+		defaultRequestWaitLimit: defaultRequestWaitLimit,
+		newCtxFn:                context.WithDeadline,
+	}
+}
+
+type requestWaitContextFactory struct {
```

Comment:
For me the factory is an overkill here.
I would switch to simple helper function `contextWithDeadline` (or sth like that) - this should make is similarly testable and would be easier to follow.

Reply:
I agree.  The only thing that is needed to be a variable here is the default wait limit.  The method here could be a helper function that takes a clock and a parent context and, if you really want to test to the max, the `newCtxFn`.  @wojtek-t: I really prefer black-box unit tests over checksum unit tests.  Unfortunately, `context.Context` does not have a way to enumerate over all the things that `Context.Value` does, so there is no way to write a black-box test that a Context transformer preserved all the `Context.Value` behavior.

Reply:
> So I would be fine with simplifying to a black-box test that examines only the Context's Deadline and Err and apirequest.ReceivedTimestampFrom.

Yeah - that was exactly where I wanted to get to.